flink-table/flink-table-common/src/main/java/org/apache/flink/table/expressions/BuiltInFunctionDefinitions.java

```diff
@@ -324,6 +324,8 @@
 		new FunctionDefinition("in", SCALAR_FUNCTION);
 	public static final FunctionDefinition CAST =
 		new FunctionDefinition("cast", SCALAR_FUNCTION);
+	public static final FunctionDefinition REINTERPRET_CAST =
```

Comment:
Hi @dawidwys 
I add a `REINTERPRET_CAST` to `BuiltInFunctionDefinitions`. It corresponds to `RexBuilder.makeReinterpretCast` for calcite.
Blink-runner supports it, while legacy-runner does not currently support it.
Look good to you?